src/Compilers/Core/Portable/InternalUtilities/SharedStopwatch.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Roslyn.Utilities
+{
+    internal struct SharedStopwatch
+    {
+        private static readonly Stopwatch s_stopwatch = Stopwatch.StartNew();
```

Comment:
There could be issues if the program is run for more than [10 million days](https://docs.microsoft.com/en-us/dotnet/api/system.timespan.maxvalue?view=netframework-4.8#remarks), but this seems like an unusual scenario.